### PR TITLE
Bacs Direct Debit component

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
Implementing the Bacs Direct Debit components (directdebit_GB)

## Tested scenarios

**Fixed issue**:  COWEB-808

### CODE REVIEW

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    width: 20%;
+
+    &-dropin{
+      top: -50px;
+    }
+
+    .adyen-checkout__bacs--edit-button{
+      border: none;
+      background: none;
+      color: blue;
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:34]: We could use here one of the colors in `colors.scss`

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      text-decoration: underline;
+      text-align: end;
+      cursor: pointer;
+    }
+  }
+}
+
+.adyen-checkout__voucher-result__introduction{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:42]:
```suggestion
.adyen-checkout__voucher-result__introduction {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    top: -25px;
+    right: 0px;
+    cursor: pointer;
+    width: 20%;
+
+    &-dropin{
+      top: -50px;
+    }
+
+    .adyen-checkout__bacs--edit-button{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:31]:
```suggestion
    .adyen-checkout__bacs--edit-button {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  .adyen-checkout__field{
+
+    &--inactive{
+      pointer-events: none;
+    }
+  }
+
+  .adyen-checkout__bacs--edit{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:19]:
```suggestion
  .adyen-checkout__bacs--edit {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    position: relative;
+
+    .adyen-checkout-input__inline-validation--valid{
+      display: none
+    }
+  }
+
+  .adyen-checkout__field{
+
+    &--inactive{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:14]:
```suggestion
    &--inactive {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+  &--confirm{
+    position: relative;
+
+    .adyen-checkout-input__inline-validation--valid{
+      display: none
+    }
+  }
+
+  .adyen-checkout__field{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:12]:
```suggestion
  .adyen-checkout__field {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 7]
```diff
@@ -0,0 +1,44 @@
+@import '../../../style';
+.adyen-checkout__bacs{
+
+  &--confirm{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:4]:
```suggestion
  &--confirm {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    position: relative;
+
+    .adyen-checkout-input__inline-validation--valid{
```
Comment (marcperez) [inline packages/lib/src/components/BacsDD/components/BacsInput.scss:7]:
```suggestion
    .adyen-checkout-input__inline-validation--valid {
```

File: packages/lib/src/components/BacsDD/components/BacsInput.scss (hunk 4 above)
Comment (sponglord) [reply packages/lib/src/components/BacsDD/components/BacsInput.scss:19]: Can the linter not be applied to .scss files?

File: packages/lib/src/components/BacsDD/components/BacsInput.scss (hunk 1 above)
Comment (sponglord) [reply packages/lib/src/components/BacsDD/components/BacsInput.scss]: done

File: packages/lib/src/components/BacsDD/components/BacsInput.scss [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      color: blue;
+      text-decoration: underline;
+      text-align: end;
+      cursor: pointer;
+    }
+  }
+}
+
+.adyen-checkout__voucher-result__introduction{
+  font-size: 0.97em;
```
[the remark this answers is not in the document]
Comment (sponglord) [reply packages/lib/src/components/BacsDD/components/BacsInput.scss]: done

File: packages/lib/src/components/ThreeDS2/ThreeDS2Challenge.tsx [hunk 10]
```diff
@@ -26,7 +26,7 @@ class ThreeDS2Challenge extends UIElement<ThreeDS2ChallengeProps> {
 
     render() {
         if (!this.props.paymentData) {
-            this.props.onError({ errorCode: 'threeds2.challenge', message: 'No paymentData received. Challenge cannot proceed' });
+            this.props.onError({ errorCode: 'threeds2.challenge', message: 'No paymentData/authorisationToken received. Challenge cannot proceed' });
```
Comment (marcperez) [inline packages/lib/src/components/ThreeDS2/ThreeDS2Challenge.tsx:29]: Is this change intentional?

File: packages/playground/src/pages/Components/Components.js [hunk 11]
```diff
@@ -54,6 +54,17 @@ getOriginKey()
...
+        window.bacsdd = checkout
+            .create('directdebit_GB', {
+                data: {
+                    holderName: 'Philip Dog',
+                    bankAccountNumber: '12345678',
+                    bankLocationId: '123456',
+                    shopperEmail: 'phil@ddog.co.uk'
+                }
+            })
+            .mount('.bacsdd-field');
```
Comment (marcperez) [inline packages/playground/src/pages/Components/Components.js:66]: Since we moved it to "Vouchers" maybe we can remove it from here.

File: packages/lib/src/components/ThreeDS2/ThreeDS2Challenge.tsx (hunk 10 above)
Comment (sponglord) [reply packages/lib/src/components/ThreeDS2/ThreeDS2Challenge.tsx:29]: yes

Comment (sponglord) [reply packages/lib/src/components/ThreeDS2/ThreeDS2Challenge.tsx:29]: But does only apply to the new flow - so I'll remove it for now

File: packages/playground/src/pages/Components/Components.js (hunk 11 above)
Comment (sponglord) [reply packages/playground/src/pages/Components/Components.js:66]: Except then you can't rest the full flow - 'cos you can't submit from Vouchers

Comment (sponglord) [reply packages/playground/src/pages/Components/Components.js]: removed form components playground

**Review by marcperez — APPROVED** [review]
